File: app/src/main/java/com/zaclimon/acetv/repository/SharedPreferencesRepositoryImpl.java
package com.zaclimon.acetv.repository;

import android.content.Context;
import android.content.SharedPreferences;

import com.zaclimon.acetv.util.Constants;

/**
 * Implementation of the {@link SharedPreferencesRepository} interface for Ace TV.
 *
 * @author zaclimon
 * Creation date: 25/06/17
 */

public class SharedPreferencesRepositoryImpl implements SharedPreferencesRepository {

    private SharedPreferences mSharedPreferences;
    private SharedPreferences.Editor mEditor;

    /**
     * Base constructor for this class
     * @param context is the context needed to retrieve the SharedPreferences instance.
     */
    public SharedPreferencesRepositoryImpl(Context context) {
        mSharedPreferences = context.getSharedPreferences(Constants.ACE_TV_PREFERENCES, Context.MODE_PRIVATE);
        mEditor = mSharedPreferences.edit();
    }

    /**
     * {@inheritDoc}
     * Please note that the default value is an empty string. ("")
     */
    @Override
    public String getString(String key) {
        return (getString(key, ""));
    }

    @Override
    public String getString(String key, String defaultValue) {
        return (mSharedPreferences.getString(key, defaultValue));
    }

    @Override
    public int getInt(String key) {
        return (getInt(key, 0));
    }

    @Override
    public int getInt(String key, int defaultValue) {
        return (mSharedPreferences.getInt(key, defaultValue));
    }

    @Override
    public boolean getBoolean(String key, boolean defaultValue) {
        return (mSharedPreferences.getBoolean(key, defaultValue));
    }

    @Override
    public void putString(String key, String value) {
        mEditor.putString(key, value);
    }

    @Override
    public void putBoolean(String key, boolean value) {
        mEditor.putBoolean(key, value);
    }

    @Override
    public void putInt(String key, int value) {
        mEditor.putInt(key, value);
    }

    @Override
    public void apply() {
        mEditor.apply();
    }

}

File: app/src/main/java/com/zaclimon/tsutaeru/repository/SharedPreferencesRepositoryImpl.kt
package com.zaclimon.tsutaeru.repository

import android.content.Context
import com.zaclimon.tsutaeru.util.Constants

/**
 * Implementation of the [SharedPreferencesRepository] interface for Tsutaeru.
 *
 * @author zaclimon
 */
class SharedPreferencesRepositoryImpl(context: Context) : SharedPreferencesRepository {

    private val repositoryPreferences = context.getSharedPreferences(Constants.TSUTAERU_PREFERENCES, Context.MODE_PRIVATE)
    private val repositoryEditor = repositoryPreferences.edit()

    override fun getString(key: String): String {
        return repositoryPreferences.getString(key, "")
    }

    override fun getString(key: String, defaultValue: String): String {
        return repositoryPreferences.getString(key, defaultValue)
    }

    override fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return repositoryPreferences.getBoolean(key, defaultValue)
    }

    override fun getInt(key: String): Int {
        return repositoryPreferences.getInt(key, 0)
    }

    override fun getInt(key: String, defaultValue: Int): Int {
        return repositoryPreferences.getInt(key, defaultValue)
    }

    override fun putString(key: String, value: String) {
        repositoryEditor.putString(key, value)
    }

    override fun putBoolean(key: String, value: Boolean) {
        repositoryEditor.putBoolean(key, value)
    }

    override fun putInt(key: String, value: Int) {
        repositoryEditor.putInt(key, value)
    }

    override fun apply() {
        repositoryEditor.apply()
    }
}